File: Quick_Sort/Quick_Sort.c
void quick(int *array, int first, int last);

void quick_Sort(int *array, int size)
{
    quick(array, 0, size - 1);
}

void quick(int *array, int first, int last)
{
    if(first >= last)
    {
        return;
    }

    int middle = (first + last) / 2;
    int pivot = array[middle];
    int i = first;
    int j = last;
    int temp;

    while(1)
    {
        while(array[i] < pivot)
        {
            i++;
        }

        while(array[j] > pivot)
        {
            j--;
        }

        if(i >= j)
        {
            break;
        }

        temp = array[i];
        array[i] = array[j];
        array[j] = temp;

        i++;
        j--;
    }

    quick(array, first, j);
    quick(array, j + 1, last);
}

File: quickSort/quickSort.py
def quickSort(array):
    _quickSort(array, 0, len(array) - 1)

def _quickSort(array, start, end):
    if start >= end:
        return

    pIndex = (start + end)/2
    pivot = array[pIndex]
    i = start
    j = end

    while True:
        while array[i] < pivot:
            i += 1
        while array[j] > pivot:
            j -= 1
        if i >= j:
            break 
        array[i], array[j] = array[j], array[i]
        i += 1
        j -= 1

    _quickSort(array, start, j)
    _quickSort(array, j + 1, end)